apis/apiextensions/v1/composition_types.go

```diff
@@ -253,6 +253,9 @@ type Patch struct {
 	// Policy configures the specifics of patching behaviour.
 	// +optional
 	Policy *PatchPolicy `json:"policy,omitempty"`
+
+	// Filtered holds whether this patch has been filtered
+	Filtered bool `json:"-"`
```

Comment:
Thanks @negz.
Yes, that's correct: The purpose was to cache the filtering status of the `Patch` object computed in `Patch.Apply`, and to avoid producing superfluous merge options for filtered patches. 
I was not either comfortable with introducing a field that's not to be serialized, and as I mentioned in the above comment, I have now removed this field. Instead, the current version recomputes filtering status.